packages/react-charting/src/utilities/UtilityUnitTests.test.ts

```diff
@@ -91,3 +101,1196 @@ describe('Unit test for getting colors from token and returning the theme specif
+    const xAxisParams = createXAxisParams();
+    utils.createNumericXAxis(xAxisParams, utils.ChartTypes.VerticalBarChart);
+    expect(xAxisParams.xAxisElement).toMatchSnapshot();
+  });
+
+  it('should create rounded x-axis labels when showRoundOffXTickValues is true', () => {
+    const xAxisParams = createXAxisParams({
+      domainNRangeValues: { dStartValue: 0.243, dEndValue: 0.433 },
+      showRoundOffXTickValues: true,
+    });
+    const result = utils.createNumericXAxis(xAxisParams, utils.ChartTypes.VerticalBarChart);
+    matchResult(convertXAxisResultToJson(result));
+  });
+
+  it('should render the x-axis labels correctly for specific tick size and padding values', () => {
+    const xAxisParams = createXAxisParams({ xAxistickSize: 10, tickPadding: 5 });
+    utils.createNumericXAxis(xAxisParams, utils.ChartTypes.VerticalBarChart);
+    expect(xAxisParams.xAxisElement).toMatchSnapshot();
+  });
+
+  it('should create the x-axis labels correctly for a specific number of ticks', () => {
+    const xAxisParams = createXAxisParams({ xAxisCount: 3 });
+    const result = utils.createNumericXAxis(xAxisParams, utils.ChartTypes.VerticalBarChart);
+    matchResult(convertXAxisResultToJson(result, false, xAxisParams.xAxisCount));
+  });
+
+  it('should render the x-axis labels correctly for horizontal bar chart with axis', () => {
+    const xAxisParams = createXAxisParams();
+    utils.createNumericXAxis(xAxisParams, utils.ChartTypes.HorizontalBarChartWithAxis);
+    expect(xAxisParams.xAxisElement).toMatchSnapshot();
+  });
+});
+
+describe('createDateXAxis', () => {
+  const domainNRangeValues: ICreateXAxisParams['domainNRangeValues'] = {
+    dStartValue: new Date(2021, 6, 1),
+    dEndValue: new Date(2022, 5, 30),
+  };
+
+  it('should render the x-axis labels correctly', () => {
+    const xAxisParams = createXAxisParams({ domainNRangeValues });
+    utils.createDateXAxis(xAxisParams, {});
+    expect(xAxisParams.xAxisElement).toMatchSnapshot();
+  });
+
+  it('should render the x-axis labels correctly for specific tick size and padding values', () => {
+    const xAxisParams = createXAxisParams({ domainNRangeValues, xAxistickSize: 10, tickPadding: 5 });
+    utils.createDateXAxis(xAxisParams, {});
+    expect(xAxisParams.xAxisElement).toMatchSnapshot();
+  });
+
+  it('should create the x-axis labels correctly for a specific number of ticks', () => {
+    const xAxisParams = createXAxisParams({ domainNRangeValues, xAxisCount: 12 });
+    const result = utils.createDateXAxis(xAxisParams, {});
+    matchResult(convertXAxisResultToJson(result, false, xAxisParams.xAxisCount));
+  });
+
+  it('should create the x-axis labels correctly when customDateTimeFormatter is provided', () => {
+    const xAxisParams = createXAxisParams({ domainNRangeValues });
+    const result = utils.createDateXAxis(xAxisParams, {}, undefined, undefined, undefined, (dateTime: Date) =>
+      dateTime.toDateString(),
```

Comment:
provide a more realistic formatter here so that users can also understand thee usage